partiql-coverage/README.md

```diff
@@ -0,0 +1,195 @@
+```partiql
+SELECT
+    measurement AS measurement,
+    CASE (measurement > 0)
+        WHEN true THEN 'measurement is positive!'
+        WHEN false THEN 'measurement is negative!'
+    END AS measurementDescription,
+    CASE
+        WHEN (complexity < 0) THEN 'complexity is negative!'
+        WHEN (complexity = 0) THEN 'complexity is zero!'
+        WHEN (complexity > 0) THEN 'complexity is positive!'
+    END AS complexityDescription
+FROM
+    t AS t
+WHERE
+    t.radius > 0
+GROUP BY
+    t.measurement AS measurement,
+    t.complexity AS complexity
+HAVING
+    measurement != 0;
+```
+
+In the above example, there are 14 branches.
+
+## Branch-Condition Coverage
+
+**Branch-Condition Coverage** is a metric calculated by determining how many *branch-conditions* are in a particular PartiQL statement and
+how many of those branch-conditions were executed. A branch-condition is the result of a boolean expression contained
+within a control-flow expression, such as an `IF` statement in many programming languages. Consider the following example:
+```partiql
+SELECT *
+FROM t
+WHERE
+    t.a > t.b
+    AND
+    t.b < t.c
+```
+
+In the above PartiQL statement, there is one control-flow expression (the `WHERE` clause). The control-flow expression
+results in two branches, however, there are 3 expressions dictating the result of the `WHERE` clause, specifically: the
+greater-than expression, the `AND` expression, and the less-than expression. Now, depending on what typing mode you are using,
+each one of these expressions can result in 3 or 4 outcomes (`TRUE`, `FALSE`, `NULL`, or `MISSING`).
+- If you are in `PERMISSIVE` mode, each of these expressions could potentially return `TRUE`, `FALSE`, `NULL`, or `MISSING`. In
+the above example, this would result in 12 branch-conditions.
+- If you are not in `PERMISSIVE` mode, each of these expressions could potentially return `TRUE`, `FALSE`, or `MISSING`. In
+the above example, this would result in 9 branch-conditions.
+
+## Things to Note
+
+1. While PartiQL’s current evaluator does not statically resolve the types of expressions, it is unknown whether a
+   specific branch-condition might return TRUE, FALSE, NULL, or MISSING. Therefore, the existing Evaluating Compiler will,
+   based on the typing mode, assume that all boolean expressions can return ALL possible outcomes. In the future, when
+   constant-folding and static-typing are part of the planning process, this will be mitigated.
+2. Similar to #1, as the current evaluate does not statically resolve the types of expressions, functions and variable
+   references will not be part of a Branch-Condition Coverage result. However, in the future, support for this will be added.
+3. As there is not a consensus regarding how `COALESCE` and `NULLIF` will be represented in the future evaluator,
+   these two SQL-defined functions currently do not create branches. However, this is subject to change in future releases.
+
```

Comment:
Just curious: does this mean that `COALESCE`, `NULLIF`, `WHERE`, `HAVING`, and `CASE_WHEN` are all the control flows in the scope or there are others too? maybe worthwhile having a pointer to this in the Wiki.

Reply:
Line 29 conflates what we support and don't. Perhaps would make sense to make what we support and don't out of all more explicit.